### PR TITLE
XCTUnwrap instead of guard

### PR DESCRIPTION
**Pull request by maratal**
Since the purpose of this PR is to improve readability of the tests, I've only replaced places that were overwhelmed with `guard` statements followed one after another. Also standalone cases where clause were one of a bunch mentioned in previous sentence (in most cases).

### CODE REVIEW

File: Spec/Tests/AuthTests.swift [hunk 1]
```diff
@@ -73,6 +73,11 @@ private func rsa8gTestsSetupDependencies() {
 private let authCallbackTestsOptions = AblyTests.clientOptions()
 private let jwtAndRestTestsOptions = AblyTests.clientOptions()
 private var client: ARTRest!
+private var jsonEncoder: ARTJsonLikeEncoder = {
+    let encoder = ARTJsonLikeEncoder()
+    encoder.delegate = ARTJsonEncoder()
+    return encoder
+}()
```
[the remark this answers is not in the document]
Comment (maratal) [reply Spec/Tests/AuthTests.swift:80]: Since it was used to improve readability, I think yes

**Review by lukasz-szyszkowski — APPROVED** [review]

File: Spec/Tests/AuthTests.swift [hunk 2]
```diff
@@ -2970,18 +2936,11 @@ class AuthTests: XCTestCase {
         }
     }
 
-    func test__107__authorize__should_adhere_to_all_requirements_relating_to__authUrl_with_json() {
-        guard let tokenDetails = getTestTokenDetails() else {
-            XCTFail("TokenDetails is empty")
-            return
-        }
-
-        let encoder = ARTJsonLikeEncoder()
```
Comment (lawrence-forooghian) [inline Spec/Tests/AuthTests.swift:2991]: I was wondering how this change (extracting the `encoder` variable) is related to the rest of the changes in this commit (using `XCTUnwrap`)? Also, could you explain the motivation for the change? It introduces shared state between test cases, which is something we should be trying to _reduce_, not add, I think.

**Review by lawrence-forooghian — COMMENTED** [review]
Nice changes 👍 – just one question.

File: Spec/Tests/AuthTests.swift (hunk 2 above)
Comment (maratal) [reply Spec/Tests/AuthTests.swift:2991]: About motivation behind this change - https://github.com/ably/ably-cocoa/pull/1258#discussion_r775505729
Since `ARTJsonLikeEncoder`/`ARTJsonEncoder` doesn't keep any state by themselves and are merely just a wrappers around `NSJSONSerialization` and `NSDictionary` I think it's fine to hide them behind this global var. WDYT @lukasz-szyszkowski  @QuintinWillison

File: Spec/Tests/AuthTests.swift [hunk 3]
```diff
@@ -3045,19 +2992,8 @@ class AuthTests: XCTestCase {
...
-        do {
-            encodedTokenRequest = try encoder.encode(tokenRequest)
-        } catch {
-            fail("Encode failure: \(error)")
-            return
-        }
-        guard let tokenRequestJSON = String(data: encodedTokenRequest, encoding: .utf8) else {
-            XCTFail("JSON Token Request is empty")
-            return
-        }
```
Comment (lukasz-szyszkowski) [inline Spec/Tests/AuthTests.swift:3060]: I approved this PR before but I have doubts looking at this?
1. it should be changed in a separate PR
2. don't we need to check if we can encode `tokenRequest` anymore?

File: Spec/Tests/AuthTests.swift (hunk 2 above)
Comment (lukasz-szyszkowski) [reply Spec/Tests/AuthTests.swift:2991]: [Here](https://github.com/ably/ably-cocoa/pull/1258#discussion_r780922925) is my comment

Comment (lawrence-forooghian) [reply Spec/Tests/AuthTests.swift:2991]: I think it still adds mental overhead, somebody has to know that those classes don't have any internal state, and it's an undocumented assumption which we have to make sure we don't violate by accident. I don't think it costs much to just create a new one each time:

```swift
private func createJsonEncoder() -> ARTJsonLikeEncoder {
     let encoder = ARTJsonLikeEncoder()
     encoder.delegate = ARTJsonEncoder()
     return encoder
 }
```

and in tests

```swift
let jsonTokenDetails = try XCTUnwrap(createJsonEncoder().encode(testTokenDetails), "Invalid TokenDetails")
```

I think that if we didn't already have loads of shared state in the tests, it would bother me less, but since doing #1240 we have a lot already.

File: Spec/Tests/AuthTests.swift (hunk 3 above)
Comment (maratal) [reply Spec/Tests/AuthTests.swift:3060]: > don't we need to check if we can encode tokenRequest anymore?
Could you elaborate on this? I mean maybe you've seen changes in the docs related to this or some other hints.

Comment (maratal) [reply Spec/Tests/AuthTests.swift:3060]: With this [comment](https://github.com/ably/ably-cocoa/pull/1258#discussion_r784825822) in mind can we agree that collapsing these lines into `XCTUnwrap` here can go into this PR as well?

Comment (lukasz-szyszkowski) [reply Spec/Tests/AuthTests.swift:3060]: My bad 🙈 it's correct, I found it here @maratal

**Review by lukasz-szyszkowski — APPROVED** [review]

**Review by lawrence-forooghian — APPROVED** [review]
